src/mobu/services/business/notebookrunner.py

```diff
@@ -68,18 +68,18 @@ def annotations(self, cell_id: str | None = None) -> dict[str, str]:
         return result
 
     async def startup(self) -> None:
-        if not self._repo:
-            self.clone_repo()
+        if not any(Path(self._repo_dir.name).rglob("*")):
+            # https://stackoverflow.com/questions/25675352/
+            await self.clone_repo()
```

Comment:
This seems unnecessarily confusing, so much so that it requires a comment pointing to a StackOverflow answer. Why not stick with the existing pattern where the Git storage is created during `startup` if it doesn't already exist, and use that as the trigger to clone the repository?

Reply:
At least conceptually, sure.  I didn't go back to the main branch, but we only set self._repo_dir in startup, and then only if the value is None.  Then we remove it in shutdown (and set it back to None, not that it matters, since the Business object is about to go away).